functions/go/set-image/custom/custom.go

```diff
@@ -0,0 +1,43 @@
+package custom
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/kustomize/api/filters/imagetag"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filtersutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func SetAdditionalFieldSpec(image types.Image, objects fn.KubeObjects, additionalImageFields types.FsSlice, ctx *fn.Context) {
+	for i, obj := range objects {
+		objRN, err := yaml.Parse(obj.String())
+		if err != nil {
+			ctx.ResultInfo(err.Error(), obj)
+		}
+		filter := imagetag.Filter{
+			ImageTag: image,
+			FsSlice:  additionalImageFields,
+		}
+
+		filter.WithMutationTracker(mutationTracker(ctx, obj))
+		err = filtersutil.ApplyToJSON(filter, objRN)
+		if err != nil {
+			ctx.ResultInfo(err.Error(), obj)
+		}
+		newObj, err := fn.ParseKubeObject([]byte(objRN.MustString()))
+		if err != nil {
+			ctx.ResultInfo(err.Error(), obj)
+		}
+		objects[i] = newObj
+	}
+}
+
+func mutationTracker(ctx *fn.Context, ko *fn.KubeObject) func(key, value, tag string, node *yaml.RNode) {
```

Comment:
Suggest renaming this function to "LogResultCallback". 
Reasons:
- The function names `mutationTracker` and `WithMutationTracker` give redundant information. 
- In general a function should serve at least one of the following goals:
   - explain what the code does. `CreateXX` `ResetXX`
   - follow some name conventions. e.g. `Request` `Response` `Get` in a http library.
  the goal is to give reviewers an idea how closely they need to dig into the code.